src/iosrv/init.c

```diff
@@ -80,7 +80,13 @@ int			dss_core_depth;
 int			dss_core_nr;
 /** start offset index of the first core for service XS */
 int			dss_core_offset;
-
+/** NUMA node to bind to */
+int			numa_node;
```

Comment:
By design, it is limited to bind all service XS to one NUMA node.  If there are 4 NUMA nodes and two servers, each server can use exactly one NUMA node each.  The reason for binding to the NUMA node is to obtain the benefits of spatial locality between the network device, memory, and cores.  If we allow binding to cores that are outside of the given NUMA node, the performance benefit is diminished, and there will be different performance among XS threads for the same server.

Reply:
hmm, by honest I don't think this is a reasonable design. Especially for the case that there is multiple NUMA nodes and one NIC, use even cannot just run a daos server and use all CPU cores after this patch.
I think we need not to limit daos server's resource to just one NUMA node, i.e daos server should be able to use multiple nodes' resource.

For now one acceptable method could be that define "int dss_numa_node = -1" that by default it is -1 if no that option specified, in that case just bypass the numa binding code and daos_server can still use all NUMA nodes same as before. This needs minimal change of your current patch.

Reply:
In the A21 system, the design has exactly two NICs, with each connected to a dedicated NUMA node.  The idea of binding cores to a specific and single NUMA node follows from that.  I understand that other system architectures unrelated to A21 will look differently.  If we want to make it possible to bind to multiple NUMA nodes at the same time or to have even more flexibility on the core binding, we can pass in the allowable core set as you wrote in the comments, or we can pass in a bit mask that represents the CPU set bitmask that hwloc uses.  Since this is a design change, I want feedback from @johannlombardi to see what he wants to do here.